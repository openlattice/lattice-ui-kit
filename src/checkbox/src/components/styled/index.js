// @flow
import styled, { css } from 'styled-components';
import { NEUTRALS, PURPLES } from '../../../../colors';

const CheckboxLabel = styled.label`
  align-items: center;
  color: ${NEUTRALS[0]};
  display: inline-flex;
  font-size: 14px;
  font-weight: normal;
  margin: 10px 0;
  min-height: 20px;
  padding: 0 10px 0 30px;
  pointer-events: ${props => (props.readOnly ? 'none' : 'auto')};
  position: relative;
`;


type IndicatorProps = {
  checked :boolean;
};

const getAfterStyles = ({ checked } :IndicatorProps) => {
  if (!checked) {
    return css`
      display: none;
    `;
  }

  return css`
    content: '';
    position: absolute;
    left: 8px;
    top: 4px;
    width: 3px;
    height: 8px;
    border: solid white;
    border-width: 0 2px 2px 0;
    transform: rotate(45deg);
  `;
};

const CheckboxIndicator = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  height: 20px;
  width: 20px;
  background: ${PURPLES[6]};
  border-radius: 3px;

  :after {
    ${getAfterStyles}
    }
  }
`;

const CheckboxInput = styled.input.attrs(() => ({
  type: 'checkbox'
}))`
  position: absolute;
  z-index: -1;
  opacity: 0;
  vertical-align: middle;
  
  :hover + ${CheckboxIndicator},
  :focus + ${CheckboxIndicator} {
    background: ${NEUTRALS[1]};
  }

  :checked + ${CheckboxIndicator} {
    background: ${PURPLES[2]};
  }

  :checked:hover + ${CheckboxIndicator},
  :checked:focus + ${CheckboxIndicator} {
    background: ${PURPLES[1]};
  }

  :disabled + ${CheckboxIndicator} {
    background: ${PURPLES[6]};
    cursor: not-allowed;
  }

  :checked:disabled + ${CheckboxIndicator} {
    background: ${NEUTRALS[2]};
  }
`;


export {
  CheckboxIndicator,
  CheckboxInput,
  CheckboxLabel,
};
