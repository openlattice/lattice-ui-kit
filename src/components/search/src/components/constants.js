import { List, Map, fromJS } from 'immutable';

const mockFilterFields = [
  {
    filterCallback: (searchResult, filter) => {
      // if (filter.value === 'All') return true;
      // return searchResult.getIn([filter.id, 0]) === filter.value;
      return true;
    },
    id: 'reportType',
    label: 'Report type',
    options: ['Crisis Template', 'Follow-up'],
  },
  {
    filterCallback: (searchResult, filter) => {
      // if (filter.value === 'All') return true;
      // return searchResult.getIn([filter.id, 0]) === filter.value;
      return true;
    },
    id: 'badges',
    label: 'Badges',
    options: ['Officer Safety', 'Substance use'],
  },
  {
    filterCallback: (searchResult, filter) => {
      // if (filter.value === 'All') return true;
      // return searchResult.getIn([filter.id, 0]) === filter.value;
      return true;
    },
    id: 'submitter',
    label: 'Submitter',
    options: [
      'squidward@bubblebowl.com',
      'smitty@werbenjagermanjensen.com',
      'wumbology@thestudyofwumbo.com',
      'enigma@innermachinations.com'
    ],
  },
];

const mockSearchResultsForReports = List([
  Map({
    reportType: 'Crisis Template',
    badges: ['Officer Safety'],
    submitter: 'squidward@bubblebowl.com'
  }),
  Map({
    reportType: 'Follow-up',
    badges: ['Officer Safety', 'Substance use'],
    submitter: 'squidward@bubblebowl.com',
  }),
  Map({
    reportType: 'Crisis Template',
    badges: ['Officer Safety', 'Substance use'],
    submitter: 'smitty@werbenjagermanjensen.com'
  }),
  Map({
    reportType: 'Follow-up',
    badges: ['Substance use'],
    submitter: 'wumbology@thestudyofwumbo.com'
  }),
  Map({
    reportType: 'Crisis Template',
    badges: ['Officer Safety', 'Substance use'],
    submitter: 'enigma@innermachinations.com'
  }),
  Map({
    reportType: 'Follow-up',
    submitter: 'enigma@innermachinations.com'
  }),
]);

const mockSearchResultsForPeople = List([
  Map({
    lastName: 'Werbenjagermanjensen',
    firstName: 'Smitty',
    middleName: '',
    sex: 'M',
    gender: 'M',
    ethnicity: 'Fish',
    dob: '02/22/2002',
    identifier: '365ba027-0c71-47d1-a5f4-61346ed96d8c',
  }),
  Map({
    lastName: 'Thornton',
    firstName: 'Cameron',
    middleName: 'Orville',
    sex: 'M',
    gender: 'M',
    ethnicity: 'Black / African American',
    dob: '01/05/1982',
    identifier: '565ba027-0c71-47d1-a5f4-61346ed96d8c',
  }),
  Map({
    lastName: 'Gerner',
    firstName: 'Timothy',
    middleName: 'Pernell',
    sex: 'M',
    gender: 'M',
    ethnicity: 'Native Hawaiian or Other Pacific Islander',
    dob: '10/02/1991',
    identifier: '465ba027-0c71-47d1-a5f4-61346ed96d8c',
  })
]);

export {
  mockFilterFields,
  mockSearchResultsForReports,
  mockSearchResultsForPeople
};
