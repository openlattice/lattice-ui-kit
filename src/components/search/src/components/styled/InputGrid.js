import styled, { css } from 'styled-components';

const getGridTemplateColumns = ({ columns = 4 }) => css`
  grid-template-columns: repeat(${columns}, minmax(100px, 1fr));
`;

const InputGrid = styled.div`
  display: grid;
  grid-gap: 20px 30px;
  align-items: flex-end;
  ${getGridTemplateColumns}
`;

export default InputGrid;
