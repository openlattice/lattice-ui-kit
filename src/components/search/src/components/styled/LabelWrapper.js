import styled from 'styled-components';

const LabelWrapper = styled.div`
  display: flex;
  flex: 1 0 auto;
  flex-direction: row;
  align-items: center;
  white-space: nowrap;
`;

export default LabelWrapper;
