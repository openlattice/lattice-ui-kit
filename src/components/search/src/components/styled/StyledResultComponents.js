// @flow
import styled, { css } from 'styled-components';

const ResultGrid = styled.div`
  display: grid;
  flex: 1;
  grid-auto-flow: row;
  grid-template-columns: ${(props) => props.columns && css`repeat(${props.columns}, minmax(0, 1fr))`};
  grid-gap: 20px 30px;
`;

const ResultWrapper = styled.div`
  display: flex;
  flex-direction: row;
  padding: 10px;
`;

const ResultDetails = styled.div`
  padding: 10px 30px;
  flex: 1;
`;

const Truncated = styled.div`
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`;

const Image = styled.img`
  height: 100%;
  width: 100px;
`;

export {
  Image,
  ResultGrid,
  ResultWrapper,
  ResultDetails,
  Truncated,
};
