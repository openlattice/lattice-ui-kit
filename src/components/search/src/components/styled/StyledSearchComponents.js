// @flow
import styled, { css } from 'styled-components';

type InputGridProps = {
  columns ? :number;
  align ? :string;
}

const getGridTemplateColumns = ({ columns = 4 } :InputGridProps) => css`
  grid-template-columns: repeat(${columns}, minmax(100px, 1fr));
`;

const getAlignItems = ({ align } :InputGridProps) => css`
  align-items: ${align || 'flex-start'};
`;

const InputGrid = styled.div`
  display: grid;
  flex: 1 1 100%;
  grid-gap: 20px 30px;
  ${getAlignItems}
  ${getGridTemplateColumns}
`;

const Title = styled.h1`
  font-size: 18px;
  font-weight: normal;
  margin: 0;
`;

export {
  InputGrid,
  Title
};
