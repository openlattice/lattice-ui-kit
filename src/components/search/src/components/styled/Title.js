import styled from 'styled-components';

const Title = styled.h1`
  font-size: 18px;
  font-weight: normal;
  margin: 0;
`;

export default Title;
