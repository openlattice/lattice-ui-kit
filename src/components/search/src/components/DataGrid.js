// @flow
import { Component } from 'react';
import { Map } from 'immutable';
import { ResultGrid, Text } from './styled/StyledResultComponents';
import Label from '../../../../label';

type Props = {
  className ? :string;
  columns ? :number;
  data :Map;
  labelMap ? :Map;
  emptyString ? :string;
  truncate ? :boolean;
};

class DataGrid extends Component<Props> {

  static defaultProps = {
    className: undefined,
    columns: 4,
    emptyString: '---',
    labelMap: undefined,
    truncate: false,
  }

  transformDataToDetailsList = () => {
    const { data, emptyString, labelMap } = this.props;

    let details;
    if (labelMap && Map.isMap(labelMap)) {
      details = labelMap.map((label :string, key :string) => Map({
        label,
        value: data.get(key, emptyString),
        key
      }));
    }
    else {
      details = data.map((value :any, key :string) => Map({
        label: key,
        value,
        key
      }));
    }

    return details.toList();
  }

  render() {
    const { className, columns, truncate } = this.props;
    const details = this.transformDataToDetailsList();

    return (
      <ResultGrid className={className} columns={columns}>
        {
          details && details.map((detail :Map, index :number) => (
            <div key={index.toString()}>
              <Label subtle>
                {detail.get('label', '')}
              </Label>
              <Text truncate={truncate}>
                {detail.get('value', '')}
              </Text>
            </div>
          ))
        }
      </ResultGrid>
    );
  }
}

export default DataGrid;
