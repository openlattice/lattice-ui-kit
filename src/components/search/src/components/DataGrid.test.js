import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { Map } from 'immutable';

import { mockSearchResultsForPeople, mockResultLabels } from './constants';
import DataGrid from './DataGrid';
import Label from '../../../../label';
import { Truncated } from './styled/StyledResultComponents';

describe('DataGrid', () => {
  describe('snapshots', () => {
    test('should match snapshot', () => {
      const data = mockSearchResultsForPeople.first();
      const wrapper = mount(<DataGrid data={data} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
    test('should match snapshot with labels', () => {
      const data = mockSearchResultsForPeople.first();
      const wrapper = mount(<DataGrid data={data} labelMap={mockResultLabels} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('render', () => {
    test('should only render data for labels with matching keys', () => {
      const data = mockSearchResultsForPeople.first();
      const customResultLabels = Map({
        firstName: 'First Name',
      });

      const wrapper = mount(<DataGrid data={data} labelMap={customResultLabels} />);
      expect(wrapper.find(Label)).toHaveLength(1);
      expect(wrapper.find(Truncated)).toHaveLength(1);

      expect(wrapper.find(Label).text()).toEqual(customResultLabels.get('firstName'));
      expect(wrapper.find(Truncated).text()).toEqual(data.get('firstName'));
    });

    test('should render all data if labelMap not provided', () => {
      const data = mockSearchResultsForPeople.first();

      const wrapper = mount(<DataGrid data={data} />);
      expect(wrapper.find(Label)).toHaveLength(data.count());
      expect(wrapper.find(Truncated)).toHaveLength(data.count());
    });
  });

});
