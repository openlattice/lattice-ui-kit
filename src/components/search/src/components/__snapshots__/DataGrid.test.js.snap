// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataGrid snapshots should match snapshot 1`] = `
.c0 {
  display: grid;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  grid-auto-flow: row;
  grid-template-columns: repeat(4,minmax(0,1fr));
  grid-gap: 20px 30px;
}

.c2 {
  word-break: break-word;
}

.c1 {
  color: #1f1f22;
  display: inline-block;
  font-size: 14px;
  font-stretch: normal;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 5px 5px 5px 0;
  color: #717175;
  font-size: 0.75rem;
  font-weight: 600;
  text-transform: uppercase;
}

<DataGrid
  columns={4}
  data={
    Immutable.Map {
      "lastName": "Werbenjagermanjensen",
      "firstName": "Smitty",
      "middleName": "",
      "sex": "M",
      "gender": "M",
      "ethnicity": "Fish",
      "dob": "02/22/2002",
      "identifier": "365ba027-0c71-47d1-a5f4-61346ed96d8c",
    }
  }
  emptyString="---"
  truncate={false}
>
  <styled.div
    columns={4}
  >
    <div
      className="c0"
    >
      <div
        key="0"
      >
        <styled.label
          subtle={true}
        >
          <label
            className="c1"
          >
            lastName
          </label>
        </styled.label>
        <styled.div
          truncate={false}
        >
          <div
            className="c2"
          >
            Werbenjagermanjensen
          </div>
        </styled.div>
      </div>
      <div
        key="1"
      >
        <styled.label
          subtle={true}
        >
          <label
            className="c1"
          >
            firstName
          </label>
        </styled.label>
        <styled.div
          truncate={false}
        >
          <div
            className="c2"
          >
            Smitty
          </div>
        </styled.div>
      </div>
      <div
        key="2"
      >
        <styled.label
          subtle={true}
        >
          <label
            className="c1"
          >
            middleName
          </label>
        </styled.label>
        <styled.div
          truncate={false}
        >
          <div
            className="c2"
          />
        </styled.div>
      </div>
      <div
        key="3"
      >
        <styled.label
          subtle={true}
        >
          <label
            className="c1"
          >
            sex
          </label>
        </styled.label>
        <styled.div
          truncate={false}
        >
          <div
            className="c2"
          >
            M
          </div>
        </styled.div>
      </div>
      <div
        key="4"
      >
        <styled.label
          subtle={true}
        >
          <label
            className="c1"
          >
            gender
          </label>
        </styled.label>
        <styled.div
          truncate={false}
        >
          <div
            className="c2"
          >
            M
          </div>
        </styled.div>
      </div>
      <div
        key="5"
      >
        <styled.label
          subtle={true}
        >
          <label
            className="c1"
          >
            ethnicity
          </label>
        </styled.label>
        <styled.div
          truncate={false}
        >
          <div
            className="c2"
          >
            Fish
          </div>
        </styled.div>
      </div>
      <div
        key="6"
      >
        <styled.label
          subtle={true}
        >
          <label
            className="c1"
          >
            dob
          </label>
        </styled.label>
        <styled.div
          truncate={false}
        >
          <div
            className="c2"
          >
            02/22/2002
          </div>
        </styled.div>
      </div>
      <div
        key="7"
      >
        <styled.label
          subtle={true}
        >
          <label
            className="c1"
          >
            identifier
          </label>
        </styled.label>
        <styled.div
          truncate={false}
        >
          <div
            className="c2"
          >
            365ba027-0c71-47d1-a5f4-61346ed96d8c
          </div>
        </styled.div>
      </div>
    </div>
  </styled.div>
</DataGrid>
`;

exports[`DataGrid snapshots should match snapshot with labels 1`] = `
.c0 {
  display: grid;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  grid-auto-flow: row;
  grid-template-columns: repeat(4,minmax(0,1fr));
  grid-gap: 20px 30px;
}

.c2 {
  word-break: break-word;
}

.c1 {
  color: #1f1f22;
  display: inline-block;
  font-size: 14px;
  font-stretch: normal;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 5px 5px 5px 0;
  color: #717175;
  font-size: 0.75rem;
  font-weight: 600;
  text-transform: uppercase;
}

<DataGrid
  columns={4}
  data={
    Immutable.Map {
      "lastName": "Werbenjagermanjensen",
      "firstName": "Smitty",
      "middleName": "",
      "sex": "M",
      "gender": "M",
      "ethnicity": "Fish",
      "dob": "02/22/2002",
      "identifier": "365ba027-0c71-47d1-a5f4-61346ed96d8c",
    }
  }
  emptyString="---"
  labelMap={
    Immutable.Map {
      "lastName": "Last name",
      "firstName": "First name",
      "middleName": "Middle name",
      "sex": "Sex",
      "gender": "Gender",
      "ethnicity": "Ethnicity",
      "dob": "DOB",
      "identifier": "Identifier",
    }
  }
  truncate={false}
>
  <styled.div
    columns={4}
  >
    <div
      className="c0"
    >
      <div
        key="0"
      >
        <styled.label
          subtle={true}
        >
          <label
            className="c1"
          >
            Last name
          </label>
        </styled.label>
        <styled.div
          truncate={false}
        >
          <div
            className="c2"
          >
            Werbenjagermanjensen
          </div>
        </styled.div>
      </div>
      <div
        key="1"
      >
        <styled.label
          subtle={true}
        >
          <label
            className="c1"
          >
            First name
          </label>
        </styled.label>
        <styled.div
          truncate={false}
        >
          <div
            className="c2"
          >
            Smitty
          </div>
        </styled.div>
      </div>
      <div
        key="2"
      >
        <styled.label
          subtle={true}
        >
          <label
            className="c1"
          >
            Middle name
          </label>
        </styled.label>
        <styled.div
          truncate={false}
        >
          <div
            className="c2"
          />
        </styled.div>
      </div>
      <div
        key="3"
      >
        <styled.label
          subtle={true}
        >
          <label
            className="c1"
          >
            Sex
          </label>
        </styled.label>
        <styled.div
          truncate={false}
        >
          <div
            className="c2"
          >
            M
          </div>
        </styled.div>
      </div>
      <div
        key="4"
      >
        <styled.label
          subtle={true}
        >
          <label
            className="c1"
          >
            Gender
          </label>
        </styled.label>
        <styled.div
          truncate={false}
        >
          <div
            className="c2"
          >
            M
          </div>
        </styled.div>
      </div>
      <div
        key="5"
      >
        <styled.label
          subtle={true}
        >
          <label
            className="c1"
          >
            Ethnicity
          </label>
        </styled.label>
        <styled.div
          truncate={false}
        >
          <div
            className="c2"
          >
            Fish
          </div>
        </styled.div>
      </div>
      <div
        key="6"
      >
        <styled.label
          subtle={true}
        >
          <label
            className="c1"
          >
            DOB
          </label>
        </styled.label>
        <styled.div
          truncate={false}
        >
          <div
            className="c2"
          >
            02/22/2002
          </div>
        </styled.div>
      </div>
      <div
        key="7"
      >
        <styled.label
          subtle={true}
        >
          <label
            className="c1"
          >
            Identifier
          </label>
        </styled.label>
        <styled.div
          truncate={false}
        >
          <div
            className="c2"
          >
            365ba027-0c71-47d1-a5f4-61346ed96d8c
          </div>
        </styled.div>
      </div>
    </div>
  </styled.div>
</DataGrid>
`;
